src/app-layer-parser.c

```diff
@@ -1031,8 +1045,10 @@ int AppLayerParserGetStateProgressCompletionStatus(AppProto alproto,
 {
     SCEnter();
     int r = 0;
-    r = alp_ctx.ctxs[FLOW_PROTO_DEFAULT][alproto].
-                StateGetProgressCompletionStatus(direction);
+    if (alproto != ALPROTO_UNKNOWN) {
```

Comment:
can we avoid this here? It's used during packet runtime but it should never be true for traffic we care about in this API.

Reply:
Will do.